internal/exec/engine.go

```diff
@@ -82,11 +82,14 @@ func (e *Engine) acquireConfig() (cfg types.Config, err error) {
 	// First try read the config @ e.ConfigCache.
 	b, err := ioutil.ReadFile(e.ConfigCache)
 	if err == nil {
-		if err = json.Unmarshal(b, &cfg); err != nil {
-			e.Logger.Crit("failed to parse cached config: %v", err)
+		if err = json.Unmarshal(b, &cfg); err == nil {
+			e.client = util.NewHttpClientWithTimeouts(e.Logger, cfg.Ignition.Timeouts)
+			return
+		} else {
```

Comment:
Eh, I like explicitly handling both paths. I know it's not idiomatic though. Lemme "fix" it.